playpen/coverage/coverage_hook

```diff
@@ -200,6 +206,75 @@ def _patch_entry_module(install):
                 file.writelines(lines)
 
 
+def _patch_pulp_resource_manager_service(install):
+    """Patch the pulp_resource_manager.service file.
+
+    :param install: boolean that indicates if the patch should be installed or
+        uninstalled, ``True`` or ``False`` respectively.
+    """
+    if install:
+        if _DISTRIBUTION == 'redhat' and _DISTRIBUTION_MAJOR_VERSION == '6':
+            subprocess.check_call([
+                'sed', '-i',
+                '/^VERSION=/ a CELERYD_OPTS="-P solo"',
+                '/etc/rc.d/init.d/pulp_workers',
+            ])
+        else:
+            subprocess.check_call([
+                'sed', '-i',
+                r's/\(--heartbeat-interval=[0-9]\+\).*$/\1 -P solo/',
+                '/etc/systemd/system/pulp_resource_manager.service',
+            ])
+    else:
+        if _DISTRIBUTION == 'redhat' and _DISTRIBUTION_MAJOR_VERSION == '6':
+            subprocess.check_call([
+                'sed', '-i',
+                '/^CELERYD_OPTS="-P solo"/ d',
+                '/etc/rc.d/init.d/pulp_workers',
+            ])
+        else:
+            subprocess.check_call([
+                'sed', '-i',
+                r's/\(--heartbeat-interval=[0-9]\+\).*$/\1/',
+                '/etc/systemd/system/pulp_resource_manager.service',
+            ])
+    if _DISTRIBUTION == 'redhat' and _DISTRIBUTION_MAJOR_VERSION != '6':
+        subprocess.check_call(['systemctl', 'daemon-reload'])
+
+
+def _patch_pulp_workers_service(install):
+    """Patch the pulp_workers service file template.
+
+    :param install: boolean that indicates if the patch should be installed or
+        uninstalled, ``True`` or ``False`` respectively.
+    """
+    mod = None
+    msg = None
+    try:
+        mod = importlib.import_module('pulp.server.async.manage_workers')
+        mod = mod.__file__.rstrip('co')
+    except ImportError:
+        msg = 'Module {} import failed'.format('pulp.server.async.manage_workers')
+    if msg:
+        print >>sys.stderr, msg + ', coverage reporting may be incomplete.'
```

Comment:
This logic could be simplified to:

```python
    try:
        mod = importlib.import_module('pulp.server.async.manage_workers')
    except ImportError:
        print >>sys.stderr, 'Module {} import failed, coverage reporting may be incomplete.'.format('pulp.server.async.manage_workers')
        return
    mod = mod.__file__.rstrip('co')
    with open(mod) as file:
        ...
```

This is more of a coding style suggestion than anything else. Do what you like.